### DISCUSSION TITLE
BluetoothSerial

### DISCUSSION THREAD

**Discussion by utya** [arendst/Tasmota#12666] [Ideas]
Hi, i want to add blueooth serial bridge in esp32.
I want to add commands:
btConnect
btSetPin
btSend
I want to do it by the example of telegram, but I can not find where it is handling and paring command in console? Which class is responsible for this?

**Comment by Jason2866** [comment]
There is no general Bluetooth support. Tasmota supports [BLE scanning](https://tasmota.github.io/docs/Bluetooth/)

**Reply by utya** [reply]
i understand, i want bluetooth serial

**Reply by Jason2866** [reply]
Not implemented. Needs "someone" who does. ;-)

**Reply by utya** [reply]
i want to add, but I still cannot find where to parse commands, console handler

**Reply by barbudor** [reply]
Command parsing is not a problem. It's already done by Tasmota and you have to provide a string of commands and an array of function pointers.
There tenth of example in almost every tasmota module. 

But as Jason said the problem is adding Bluetooth in Tasmota.

**Reply by utya** [reply]
> Command parsing is not a problem. It's already done by Tasmota and you have to provide a string of commands and an array of function pointers.
> There tenth of example in almost every tasmota module.
> 
> But as Jason said the problem is adding Bluetooth in Tasmota.

Why problem in Bluetooth? I want to add using this example https://randomnerdtutorials.com/esp32-bluetooth-classic-arduino-ide/

**Reply by barbudor** [reply]
It's not because you have  a basic 2 lines sample code that you can integrate that into tasmota without breaking everything and still get wifi and Bluetooth together.
Both wifi and Bluetooth are using the same radio hardware and can't work side by side without some work.

**Reply by utya** [reply]
> It's not because you have a basic 2 lines sample code that you can integrate that into tasmota without breaking everything and still get wifi and Bluetooth together.
> Both wifi and Bluetooth are using the same radio hardware and can't work side by side without some work.

ok, i understood bluetooth (classic) and wifi  cant work together at same time.
